crowdsourcehinter/crowdsourcehinter.py

```diff
@@ -0,0 +1,352 @@
+        """
+        This view renders the hint view to the students. The HTML has the hints templated
+        in, and most of the remaining functionality is in the JavaScript.
+        """
+        html = self.resource_string("static/html/crowdsourcehinter.html")
+        frag = Fragment(html)
+        frag.add_javascript_url('//cdnjs.cloudflare.com/ajax/libs/mustache.js/0.8.1/mustache.min.js')
+        frag.add_css(self.resource_string("static/css/crowdsourcehinter.css"))
+        frag.add_javascript(self.resource_string("static/js/src/crowdsourcehinter.js"))
+        frag.initialize_js('CrowdsourceHinter', {'hinting_element': self.Element, 'isStaff': self.xmodule_runtime.user_is_staff})
+        return frag
+
+    @XBlock.json_handler
+    def get_hint(self, data, suffix=''):
+        """
+        Returns hints to students. Hints with the highest rating are shown to students unless the student has already
+        submitted the same incorrect answer previously.
+        Args:
+          data['submittedanswer']: The string of text that the student submits for a problem.
+        returns:
+          'BestHint': the highest rated hint for an incorrect answer
+                        or another random hint for an incorrect answer
+                        or 'Sorry, there are no hints for this answer.' if no hints exist
+          'StudentAnswer': the student's incorrect answer
+        """
+        # populate hint_database with hints from initial_hints if there are no hints in hint_database.
+        # this probably will occur only on the very first run of a unit containing this block.
+        for answers in self.initial_hints:
+            if answers not in self.hint_database:
+                self.hint_database[answers] = {}
+            if self.initial_hints[answers] not in self.hint_database[answers]:
+                self.hint_database[answers].update({self.initial_hints[answers]: 0})
+        answer = str(data["submittedanswer"])
+        # put the student's answer to lower case so that differences in capitalization don't make
+        # different groups of hints. this is sloppy and the execution should probably be changed.
+        answer = answer.lower()
+        found_equal_sign = 0
+        remaining_hints = int(0)
+        best_hint = ""
+        # the string returned by the event problem_graded is very messy and is different
+        # for each problem, but after all of the numbers/letters there is an equal sign, after which the
+        # student's input is shown. I use the function below to remove everything before the first equal
+        # sign and only take the student's actual input.
+        #
+        # TODO: figure out better way to directly get text of student's answer
+        if "=" in answer:
+            if found_equal_sign == 0:
+                found_equal_sign = 1
+                eqplace = answer.index("=") + 1
+                answer = answer[eqplace:]
+        remaining_hints = str(self.find_hints(answer))
```

Comment:
Minor: have find_hints return booleans (True / False), and don't convert to strings.